ast2vec/cloning.py

```diff
@@ -0,0 +1,213 @@
+        return classified
+
+    def cleanup_repo(self, classified: dict, repo_dir: str) -> None:
+        """
+        Delete files not classified by enry (if some languages were specified) then only them
+        are preserved.
+
+        :param classified: enry output loaded from JSON.
+        :param repo_dir: Path to repository directory.
+        :return:
+        """
+        if self._languages:
+            self._log.info("Removing files with languages not in (%s)",
+                           ",".join(self._languages))
+            languages = self._languages
+        else:
+            self._log.info("Removing files not classified by enry.")
+            languages = classified.keys()
+
+        allowed_files = set(str.encode(os.path.join(repo_dir, fname)) for lang in languages
+                            if lang in classified for fname in classified[lang])
+
+        for dirname, dirnames, filenames in os.walk(str.encode(repo_dir)):
+            for filename in filenames:
+                full_filename = os.path.join(dirname, filename)
+                if full_filename not in allowed_files:
+                    os.remove(full_filename)
+
+    def clone_repos(self, inputs, output, ignore):
+        with Pool(self._num_threads) as pool:
+            pool.map(partial(self.process_repo, ignore=ignore, target_dir=output),
+                     self.generate_repo_urls(inputs))
+
+    def process_repo(self, git_url: str, ignore: bool, target_dir: str) -> None:
+        repo_dir = self.clone_repo(git_url, ignore, target_dir)
+        if repo_dir is None or self._linguist is None:
+            return
+        classified = self.classify_repo(repo_dir)
+        if classified:
+            self.cleanup_repo(classified, repo_dir)
+
+    @staticmethod
+    def generate_repo_urls(inputs):
+        """
+        Parse provided inputs.
+
+        :param inputs: list of files and/or git urls
+        :return: generator of git urls
+        """
+        for item in inputs:
+            if os.path.isfile(item):
+                with open(item, encoding="utf8") as f:
+                    for line in f:
+                        yield line.rstrip()
+            else:
+                yield item
+
+    def _find_enry(self, linguist):
+        if linguist is None:
+            if self._languages:
+                self._log.info("Looking for enry to work with provided languages.")
+                linguist = shutil.which("enry", path=os.getcwd())
+                if linguist is None:
+                    linguist = "enry"
+            else:
+                return None, False
+        full_path = shutil.which(linguist)
+        if not full_path:
+            raise FileNotFoundError("%s was not found. Install it: python3 -m ast2vec enry" %
+                                    linguist)
+        with open(full_path, "rb") as fin:
+            return linguist, fin.read(15) != b"#!/usr/bin/ruby"
```

Comment:
May you add comment about logic `fin.read(15) != b"#!/usr/bin/ruby"`?
or it's better to create parameter/attribute/etc instead of magic variable `b"#!/usr/bin/ruby"` (related to linguist)